solutions/instances/main.tf

```diff
@@ -4,6 +4,8 @@
 
 locals {
 
+  # tflint-ignore: terraform_unused_declarations
+  validate_existing_kms_inputs = (var.existing_cos_kms_key_crn != null && !var.skip_cos_kms_auth_policy) ? (var.existing_kms_instance_crn == null ? tobool("The existing_kms_instance_crn is not provided and is required to configure the COS - KMS authorization policy") : true) : true
```

Comment:
If a consumer if using all their own buckets, then `existing_kms_instance_crn` isnt needed right? You wont need to create any KMS auth policy because your not encrypting any buckets.

Reply:
yes, `existing_kms_instance_crn` is not required in that case. Then, `existing_cos_kms_key_crn` will be null so this validation will work